src/NServiceBus.Core/Unicast/Behaviors/MessageHandler.cs

```diff
@@ -7,14 +7,75 @@
     /// </summary>
     public class MessageHandler
     {
+        Action<object, object, object> invocationWithContext;
+
+        /// <summary>
+        /// Creates a new instance of the message handler
+        /// </summary>
+        [ObsoleteEx(TreatAsErrorFromVersion = "6", RemoveInVersion = "7", ReplacementTypeOrMember = "MessageHandler(Action<object, object, object> invocationWithContext, Type handlerType)")]
```

Comment:
what do we do when users want to test a behavior which is using the MessageHandler class. If I don't expose the `invocationWithContext` delegate that class is not usable by end users without the real pipeline. Could that be a problem?